lucet-wasi/src/memory.rs

```diff
@@ -397,6 +419,14 @@ dec_enc_scalar!(
     enc_timestamp_byref
 );
 
+pub fn dec_u32(x: u32) -> u32 {
+    u32::from_le(x)
+}
+
+pub fn enc_u32(x: u32) -> u32 {
+    x
```

Comment:
It's surprising that `enc_u32` isn't the inverse of `dec_u32` here. I don't use a use of `dec_u32` in the patch though.

On a tangential note, I'm starting to think that the "wasm32.rs" vs "host.rs" split (that I'm partly to blame for) is a little confusing, especially with all the `enc`/`dec`/ functions for scalar types that are effectively no-ops. I think we can unify things when they're defined to be fixed-size types that are the same on both sides. But that's a bigger topic that we can figure out later :-).

Reply:
I guess this is due to the fact that WASM is always assumed to be little-endian.

Reply:
Actually, the `dec_enc_scalar()` properly converts to little-endian, so I guess not doing it in `enc_u32()` is an oversight. I'm gonna add the missing `.to_le()`.

Reply:
Wasm is little-endian, but the question is where the endian translation happens. It's surprising to have functions named `dec_u32` and `enc_u32` where the former does an endian translation going one way, and the later does not do an endian translation going the other way.

Reply:
Done :)